yt/yt/server/job_proxy/unittests/stderr_writer_ut.cpp

```diff
@@ -0,0 +1,238 @@
+        writer << number;
+        reference << number;
+    }
+
+    {
+        const auto lastByte = static_cast<decltype(NApi::TGetJobStderrResponse::TotalSize)>(reference.Str().size());
+        {
+            const auto data = writer.GetCurrentData({});
+            ASSERT_EQ(data.Data.ToStringBuf(), reference.Str());
+            ASSERT_EQ(data.EndOffset, lastByte);
+            ASSERT_EQ(data.TotalSize, static_cast<decltype(data.TotalSize)>(reference.Str().size()));
+        }
+
+        {
+            const auto data = writer.GetCurrentData({.Limit = 123});
+            ASSERT_EQ(data.Data.ToStringBuf(), reference.Str().substr(0, 123));
+            ASSERT_EQ(data.EndOffset, 123);
+            ASSERT_EQ(data.TotalSize, lastByte);
+        }
+
+        {
+            const auto data = writer.GetCurrentData({.Limit = 123, .Offset = 10});
+            ASSERT_EQ(data.Data.ToStringBuf(), reference.Str().substr(10, 123));
+            ASSERT_EQ(data.EndOffset, 10 + 123);
+            ASSERT_EQ(data.TotalSize, lastByte);
+        }
+
+        {
+            const auto data = writer.GetCurrentData({.Offset = -50});
+            ASSERT_EQ(data.Data.ToStringBuf(), reference.Str().substr(reference.Str().size() - 50, 50));
+            ASSERT_EQ(data.EndOffset, lastByte);
+            ASSERT_EQ(data.TotalSize, lastByte);
+        }
+
+        {
+            // before start
+            const auto data = writer.GetCurrentData({.Offset = -50000});
+            ASSERT_EQ(data.Data.ToStringBuf(), reference.Str());
+            ASSERT_EQ(data.EndOffset, lastByte);
+            ASSERT_EQ(data.TotalSize, lastByte);
+        }
+
+        {
+            // Requested more than have
+            const auto data = writer.GetCurrentData({.Limit = 100, .Offset = 250});
+            ASSERT_EQ(data.Data.size(), size_t(44));
+            ASSERT_TRUE(data.Data.ToStringBuf().EndsWith("100\n"));
+            ASSERT_EQ(data.EndOffset, lastByte);
+            ASSERT_EQ(data.TotalSize, lastByte);
+        }
+
+        {
+            // Range after end
+            const auto data = writer.GetCurrentData({.Limit = 123, .Offset = 300});
+            ASSERT_EQ(data.Data.ToStringBuf(), "");
+            ASSERT_EQ(data.EndOffset, 0);
+            ASSERT_EQ(data.TotalSize, lastByte);
+        }
+    }
+
+    // Reach ..skipped..
+    for (size_t i = 100; i <= 1000; ++i) {
+        const auto number = ToString(i) + "\n";
+        writer << number;
+        reference << number;
+    }
+
+    {
+        const auto data = writer.GetCurrentData({});
+        ASSERT_EQ(data.TotalSize, static_cast<decltype(data.TotalSize)>(reference.Str().size()));
+    }
+
+    for (size_t i = 1000; i <= 2000; ++i) {
+        const auto number = ToString(i) + "\n";
+        writer << number;
+        reference << number;
+    }
+
+    const auto lastByte = static_cast<decltype(NApi::TGetJobStderrResponse::TotalSize)>(reference.Str().size());
+
+    {
+        // Full log requested
```

Comment:
Use dot at the end of the comment here an below in this file. Also all comments should start with capital letter as usual english sentense.